rpcs3/Emu/Cell/SPULLVMRecompiler.cpp

```diff
@@ -6114,25 +6083,16 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 			return;
 
 		// GOW 3(uses 1.0f * spu_re(div) instead of just spu_re(div) in the pattern)
-		auto check_alternative_reciprocal_pattern_for_float = [&](f32 float_value) -> bool
+		if (auto [ok_fm, div] = match_expr(c, fm(spu_re(MT), fsplat<f32[4]>(1.0f))); ok_fm)
 		{
-			if (auto [ok_fm, div] = match_expr(c, fm(spu_re(MT), fsplat<f32[4]>(1.0f))); ok_fm)
+			const auto full_expr = fma(a, b, c);
```

Comment:
It may save some lines of code but it may actually get outputted as instructions. I get what you are trying to do here but the match system is not ready for it.

Reply:
I checked emitted IR and it's all good.